Added sparc for voice cloning

fabiocat93 opened the pull request:
## Description
Added [SPARC](https://github.com/Berkeley-Speech-Group/Speech-Articulatory-Coding) model for voice cloining.

## Motivation and Context
This voice cloning model is based on Vocal Tract Kinematics (see https://arxiv.org/abs/2406.12998), which may enable preserving articulatory features as in the source voice while changing the voice texture. 

## How Has This Been Tested?
Unit tests.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.

fabiocat93 committed:
added sparc for voice cloning.

wilke0818 commented on src/senselab/audio/tasks/voice_cloning/api.py:
```diff
@@ -20,48 +21,54 @@ def clone_voices(
-    if isinstance(model, CoquiTTSModel):
+    if model is None:
+        lang = kwargs.pop("lang", None)
+        return SparcVoiceCloner.clone_voices(
```
why are we using this as a default for providing none? We should prob have a way to specify a Sparc model similar to how we do with Coqui.

wilke0818 reviewed (approved):
Looks alright. Not sure I am a big fan of how we're using Sparc as a default/no other way to specify it. Seems a bit inconsistent with the rest of the package.

fabiocat93 committed:
attempting to clean up disk space in the aws ec2 runner

fabiocat93 committed:
Update tests.yaml

changing HF_HOME to point to a different volume (and hopefully solve the out of space issue)

fabiocat93 committed:
fixing style.

fabiocat93 committed:
fixing style..

fabiocat93 committed:
the new setup with HF_HOME worked well with one job and I extended it to all jobs
